packages/core/echo/echo-db/src/core-db/core-database-query-context.ts

```diff
@@ -80,26 +90,69 @@ export class CoreDatabaseQueryContext implements QueryContext {
     queryStartTimestamp: number,
     result: RemoteQueryResult,
   ): Promise<QueryResult | null> {
-    const objectDocId = this._coreDatabase._automergeDocLoader.getObjectDocumentId(result.id);
-    if (objectDocId !== result.documentId) {
-      log("documentIds don't match", { objectId: result.id, expected: result.documentId, actual: objectDocId ?? null });
+    if (!SpaceId.isValid(result.spaceId)) {
+      log.warn('dropping result with invalid space id', { id: result.id, spaceId: result.spaceId });
       return null;
     }
 
-    const core = await this._coreDatabase.loadObjectCoreById(result.id);
-    if (!core || ctx.disposed) {
-      return null;
+    /**
+     * Ignore data in the query result and fetch documents through DataService & RepoProxy.
+     */
+    const FORCE_DATA_SERVICE_FETCH = true;
```

Comment:
not sure I understand how this is going to be used

Reply:
I'm thinking about ways to implement this where it doesn't require two separate RPCs: query to get the ideas, and then a separate document fetch. This flag is just to avoid deleting or commenting out the other branch of the code that experiments with doing a single query RPC that holds the data too. It doesn't full work because it doesn't seed the CRDT data, which means that you won't be able to mutate the queried objects later.